TikTokTrainer/Data/DataProcessing.swift

```diff
@@ -0,0 +1,141 @@
+          VNRecognizedPointKey(rawValue: "head_joint"))),
+        ((VNRecognizedPointKey(rawValue: "neck_1_joint"),
+          VNRecognizedPointKey(rawValue: "right_shoulder_1_joint"),
+          VNRecognizedPointKey(rawValue: "right_forearm_joint"))),
+        ((VNRecognizedPointKey(rawValue: "head_joint"),
+          VNRecognizedPointKey(rawValue: "neck_1_joint"),
+          VNRecognizedPointKey(rawValue: "right_shoulder_1_joint")))
+    ]
+    
+    /// Computes angles of PoseNet data with trig
+    /// Cycles through sets of joints to track which angles are available for capture, otherwise angle is marked as 0
+    ///
+    /// - Parameters:
+    ///     - video: The video uploaded by the user and processed by the PoseNetProcessor
+    private func computeAngles(video: ProcessedVideo) -> [[CGFloat]] {
+        var angles = [[CGFloat]]()
+        angles.append([])
+        var anglesIndex = 0
+
+        for slice in video.data {
+            for triple in jointTriples {
+                let pntOne = triple.0
+                let pntTwo = triple.1
+                let pntThree = triple.2
+                var angle: CGFloat = 0
+
+                if slice.points[pntOne] != nil && slice.points[pntTwo] != nil && slice.points[pntThree] != nil {
+                    angle = angleBetweenPoints(leftPoint: slice.points[pntThree]!, middlePoint: slice.points[pntTwo]!, rightPoint: slice.points[pntOne]!)
+                    print(pntTwo.rawValue + ": " + angle.description)
+                }
+                print()
+                angles[anglesIndex].append(angle)
+            }
+            angles.append([])
+            anglesIndex += 1
+        }
+        return angles
+    }
+
+    private func angleBetweenPoints(leftPoint: VNRecognizedPoint, middlePoint: VNRecognizedPoint, rightPoint: VNRecognizedPoint) -> CGFloat {
+        return angleBetweenPoints(leftPoint: CGPoint(x: leftPoint.x, y: leftPoint.y), middlePoint: CGPoint(x: middlePoint.x, y: middlePoint.y), rightPoint: CGPoint(x: rightPoint.x, y: rightPoint.y))
+    }
+
+    private func angleBetweenPoints(leftPoint: CGPoint, middlePoint: CGPoint, rightPoint: CGPoint) -> CGFloat {
+        let rightVector = (x: rightPoint.x - middlePoint.x, y: rightPoint.y - middlePoint.y)
+        let leftVector = (x: leftPoint.x - middlePoint.x, y: leftPoint.y - middlePoint.y)
+        let dotProduct = rightVector.x * leftVector.x + rightVector.y * leftVector.y
+        let determinant = rightVector.x * leftVector.y - rightVector.y * leftVector.x
+        var angle = atan2(determinant, dotProduct) * (180 / .pi)
+        if angle < 0 {
+            angle += 360
+        }
+        return angle
+    }
+
+    private func computeAngleDifferences(preRecordedVid: ProcessedVideo, recordedVid: ProcessedVideo) -> [[CGFloat]] {
+        let preRecordedPoses = computeAngles(video: preRecordedVid)
+        let recordedPoses = computeAngles(video: recordedVid)
+        var angleDifferences = [[CGFloat]]()
```

Comment:
You could declare the size of the array here instead of appending every time below